test/MHD/MTI/analysis.cpp

```diff
@@ -0,0 +1,130 @@
+#include "analysis.hpp"
+#include "idefix.hpp"
+#include "fluid.hpp"
+#include <iostream>
+#include <fstream>
+#include <iomanip>
+
+Analysis::Analysis(Input &input, Grid &grid, DataBlock &data, Output &output, std::string filename) {
+      this->d = new DataBlockHost(data);
+      this->grid = &grid;
+      this->filename = filename;
+      this->precision = 10;
+}
+
+/* **************************************************************** */
+double Analysis::Average(const int nfields, int fields[])
+    /*
+     * compute the weighted average: int dphi dz rho *infield/int dphi dz rho
+     *
+     **************************************************************** */
+{
+  real outfield = 0;
+
+  for(int k = d->beg[KDIR]; k < d->end[KDIR] ; k++) {
+    for(int j = d->beg[JDIR]; j < d->end[JDIR] ; j++) {
+      for(int i = d->beg[IDIR]; i < d->end[IDIR] ; i++) {
+        real q=1.0;
+        for(int n=0 ; n < nfields ; n++) {
+          q = q*d->Vc(fields[n],k,j,i);
+        }
+        outfield += q;
+      }
+    }
+  }
+
+    // Reduce
+#ifdef WITH_MPI
+  real reducedValue;
+  MPI_Reduce(&outfield, &reducedValue, 1, MPI_DOUBLE, MPI_SUM, 0, MPI_COMM_WORLD);
+  outfield = reducedValue;
+#endif
+
+  outfield = outfield / ((double) grid->np_int[IDIR] * grid->np_int[JDIR] * grid->np_int[KDIR]);
+
+
+  return outfield;
+}
+
+/* **************************************************************** */
+void Analysis::WriteField(double data) {
+/*
+ *  * Write a global profile to a file
+ *   *
+ *    *
+ *     **************************************************************** */
+  if(idfx::prank==0) {
+    int col_width = precision + 10;
+    this->file << std::scientific << std::setw(col_width) << data;
+  }
+  return ;
+}
+
+
+void Analysis::ResetAnalysis() {
+  GridHost gh(*this->grid);
+  gh.SyncFromDevice();
+  int col_width = precision + 10;
+  if(idfx::prank==0) {
+    file.open(filename, std::ios::trunc);
+    file << std::setw(col_width) << "t";
```

Comment:
Adding a `#` char at the beginning of the line makes this kind of csv file much easier to read via standard functions (for instance `pandas.read_csv`)
```suggestion
    file.open(filename, std::ios::trunc);
    file << "# ";
    file << std::setw(col_width) << "t";
```

Reply:
to clarify: it's not about Python (even though `pandas` is a Python library). `#` delimitates comments also in `csv`, which is a standard file format. Complying with standards improves interoperability of software, I used `pandas.read_csv` as an illustrative example but it goes beyond that.

And for completion about `pandas.read_csv` : 
- `header="infer"` is the default behaviour (the default is "assume the file is actually css-compliant")
- the function is actually flexible enough that a non-standard header (as is done currently) is also supported (using `header=0`), but it's extra effort on the user

Reply:
many analysis routines are performed the way Jean did it, and I agree that for the sake of uniformity, it's better to avoid this #. In any case, the first few lines are never csv compliant, and do require specific actions to be read. So I'm happy with the current version of the Analysis routines.

Reply:
fair enough !